### DISCUSSION TITLE
Support Gmail-style bcc header insertion (bcc recipients have their own email in bcc headers)

### DISCUSSION THREAD

**Discussion by 6peterlu** [haraka/Haraka#3430] [Ideas]
### Is your feature request related to a problem? Please describe.

Gmail will add bcc recipients to their respective MIME headers, so that each bcc recipient sees their own bcc information in the headers but not others. Haraka appears to create identical MIME headers for each recipient.

### Describe the solution you'd like

Haraka adds bcc headers for bcc recipients so that they can see themselves in the headers but not other bcc recipients.

### Describe alternatives you've considered
An alternative is to compose the MIMEs outside of Haraka and send multiple SMTP requests, one for each bcc recipient, but this seems like functionality that is ideally contained within haraka.

**Comment by msimerson** [comment]
> this seems like functionality that is ideally contained within haraka

Perhaps as a plugin. PRs welcome.

PS: there's already a long standing method of sending BCCs: the author puts themselves in the To field and all the recipients in the BCC. It works great and email users are accustomed to it.
